be/src/vec/exec/vjson_scanner.cpp

```diff
@@ -0,0 +1,729 @@
+    uint8_t tmp_buf[128] = {0};
+    int32_t wbytes = 0;
+    
+    if (slot_desc->is_nullable()) {
+        auto* nullable_column =
+            reinterpret_cast<vectorized::ColumnNullable*>(column_ptr);
+        nullable_column->get_null_map_data().push_back(0);
+        column_ptr = &nullable_column->get_nested_column();
+    }
+
+    switch (value->GetType()) {
+    case rapidjson::Type::kStringType:
+        str_value = value->GetString();
+        wbytes = strlen(str_value);
+        break;
+    case rapidjson::Type::kNumberType:
+        if (value->IsUint()) { 
+            wbytes = sprintf((char *)tmp_buf, "%u", value->GetUint());
+            str_value = (char *)tmp_buf;
+        } else if (value->IsInt()) {
+            wbytes = sprintf((char *)tmp_buf, "%d", value->GetInt());
+            str_value = (char *)tmp_buf;
+        } else if (value->IsUint64()) {
+            wbytes = sprintf((char *)tmp_buf, "%lu", value->GetUint64());
+            str_value = (char *)tmp_buf;
+        } else if (value->IsInt64()) {
+            wbytes = sprintf((char *)tmp_buf, "%ld", value->GetInt64());
+            str_value = (char *)tmp_buf;
+        } else {
+            wbytes = sprintf((char *)tmp_buf, "%f", value->GetDouble());
+            str_value = (char *)tmp_buf;
+        }
+        break;
+    case rapidjson::Type::kFalseType:
+        wbytes = 1;
+        str_value = (char *)"0";
+        break;
+    case rapidjson::Type::kTrueType:
+        wbytes = 1;
+        str_value = (char *)"1";
+        break;
+    case rapidjson::Type::kNullType:
+        if (slot_desc->is_nullable()) {
+            auto* nullable_column = reinterpret_cast<vectorized::ColumnNullable*>(column_ptr);
+            nullable_column->insert_data(nullptr, 0);
+        } else {
+            RETURN_IF_ERROR(_state->append_error_msg_to_file([&]() -> std::string { return JsonReader::_print_json_value(*value); },
+                    [&]() -> std::string {
+                    fmt::memory_buffer error_msg;
+                    fmt::format_to(error_msg, "Json value is null, but the column `{}` is not nullable.", slot_desc->col_name());
+                    return fmt::to_string(error_msg);
+                    }, _scanner_eof));
+            _counter->num_rows_filtered++;
+            *valid = false;
+            return Status::OK();
+        }
+        break;
+    default:
+        // for other type like array or object. we convert it to string to save
+        std::string json_str = JsonReader::_print_json_value(*value);
+        wbytes = json_str.size();
+        str_value = json_str.c_str();
+        break;
+    }
+    
+    RETURN_IF_ERROR(_insert_to_column(column_ptr, slot_desc, str_value, wbytes));
+
+    *valid = true;
+    return Status::OK();
+}
+
+Status VJsonReader::_insert_to_column(vectorized::IColumn* column_ptr, SlotDescriptor* slot_desc,
+                                    const char* value_ptr, int32_t& wbytes) {
+    switch (slot_desc->type().type) {
```

Comment:
the type seems always is varchar ?

Reply:
yes, VExpr just can process the input of which type is varchar